src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -754,21 +798,32 @@ async function webviewPreloads(style: PreloadStyles, rendererData: readonly Webv
 				const resources = event.data.resources;
 				let queue: Promise<PreloadResult> = Promise.resolve({ state: PreloadState.Ok });
 				for (const { uri, originalUri, source } of resources) {
-					const globals = source === 'kernel'
-						? kernelPreloadGlobals
-						: { acquireNotebookRendererApi: () => acquireNotebookRendererApi(source.rendererId) };
-
 					// create the promise so that the scripts download in parallel, but
 					// only invoke them in series within the queue
-					const promise = runScript(uri, originalUri, globals);
-					queue = queue.then(() => promise.then(fn => {
-						const result = fn();
-						if (result.state === PreloadState.Error) {
-							console.error(result.error);
-						}
 
-						return result;
-					}));
+					if (source === 'kernel') {
+						const promise = runPreload(uri, originalUri);
+						queue = queue.then(() => promise.then(async module => {
+							try {
+								await module.activate();
+								return { state: PreloadState.Ok };
+							} catch (error) {
+								console.error(error);
+								return { state: PreloadState.Error, error: error.toString() };
+							}
+						}));
```

Comment:
You'll also want to handle any errors in the rejection of `runPreload`

```suggestion
						queue = queue.then(async () => {
							try {
								const module = await promise;
								await module.activate();
								return { state: PreloadState.Ok };
							} catch (error) {
								console.error(error);
								return { state: PreloadState.Error, error: error.toString() };
							}
						}));
```